projects/github-actions/repo-gardening/src/tasks/add-labels/index.js

```diff
@@ -330,26 +331,83 @@ async function addLabels( payload, octokit ) {
 	const { owner, name } = repository;
 	const { draft, title } = pull_request;
 
+	// GitHub allows 100 labels on a PR.
+	// Limit to less than that to allow a buffer for future manual labels.
+	const maxLabels = 90;
+	const bigProjectLabel = '[Project] All the things!';
+
 	// Get labels to add to the PR.
 	const isDraft = !! ( pull_request && draft );
 
 	// If the PR title includes the word "revert", mark it as such.
 	const isRevert = title.toLowerCase().includes( 'revert' );
 
-	const labels = await getLabelsToAdd( octokit, owner.login, name, number, isDraft, isRevert );
+	const fileDerivedLabels = await getFileDerivedLabels(
+		octokit,
+		owner.login,
+		name,
+		number,
+		isDraft,
+		isRevert
+	);
+
+	// Grab current labels on the PR.
+	// We can't rely on payload, as it could be outdated by the time this runs.
+	const currentLabels = await getLabels( octokit, owner.login, name, number );
+
+	// This is an array of labels that GitHub doesn't already have.
+	let labelsToAdd = fileDerivedLabels.filter( label => ! currentLabels.includes( label ) );
 
-	if ( ! labels.length ) {
-		debug( 'add-labels: Could not find labels to add to that PR. Aborting' );
+	// Nothing new was added, so abort.
+	if ( labelsToAdd.length === 0 ) {
+		debug( 'add-labels: No new labels to add to that PR. Aborting.' );
 		return;
 	}
 
-	debug( `add-labels: Adding labels ${ labels } to PR #${ number }` );
+	// Determine how many labels can safely be added.
+	let maxLabelsToAdd = Math.max( 0, maxLabels - currentLabels.length );
+
+	// Overkill, but let's prevent this label from counting toward the max.
+	const hasBigProjectLabel = currentLabels.includes( bigProjectLabel );
+	if ( hasBigProjectLabel ) {
+		maxLabelsToAdd++;
+	}
+
+	// If there are too many labels, we need to reduce the label count to keep GitHub happy.
+	if ( labelsToAdd.length > maxLabelsToAdd ) {
+		debug( `add-labels: Too many labels! Grouping project labels into '${ bigProjectLabel }'.` );
+
+		// Filter out project-type labels in deference to bigProjectLabel.
+		// In theory we could also remove any existing project-type labels here, but for now
+		// let's not as that would prevent manually adding specific project labels.
```

Comment:
A good heuristic might be to remove any existing project labels if `! hasBigProjectLabel`, but skip any project label changing if it is.

Reply:
Then again, we'd probably get people that see the bigProjectLabel and decide to clean it up by removing that and putting a specific one. Then gardening would run and remove their changes. 😄

I'll leave it for now.